ts/Series/Treegraph/TreegraphSeries.ts

```diff
@@ -0,0 +1,659 @@
+import TU from '../TreeUtilities.js';
+const { prototype: { symbols } } = SVGRenderer;
+const { getLevelOptions } = TU;
+const {
+    seriesTypes: {
+        sankey: { prototype: sankeyProto }
+    }
+} = SeriesRegistry;
+
+import H from '../../Core/Globals.js';
+import U from '../../Core/Utilities.js';
+const { merge, pick, addEvent, stableSort } = U;
+
+import './TreegraphLayout.js';
+import SankeySeries from '../Sankey/SankeySeries.js';
+
+interface LayoutAlgorythmParameters {
+    ax: number;
+    bx: number;
+    ay: number;
+    by: number;
+}
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * treegraph series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.treegraph
+ *
+ * @augments Highcharts.Series
+ */
+class TreegraphSeries extends OrganizationSeries {
+    /* *
+     *
+     *  Static Properties
+     *
+     * */
+
+    /**
+     * A treegraph series is a diagram, which shows a relation between ancestors
+     * and descendants with a clear parent - child relation. Our treegraph
+     * algorythm accepts the data, where each point (apart from the root of the
+     * tree) has a single parent. The best examples of the dataStructures, which
+     * best reflect this chart are e.g. genealogy tree or directory scructure.
+     *
+     *
+     * @extends      plotOptions.treegraph
+     * @since        next
+     * @product      highcharts
+     * @requires     modules/sankey
+     * @requires     modules/treegraph
+     * @exclude      linkColor, linkLineWidth, linkRadius
+     * @optionparent plotOptions.treegraph
+     */
+    public static defaultOptions: TreegraphSeriesOptions = merge(
+        OrganizationSeries.defaultOptions,
+        {
+            /**
+             * The option to choose the layout alogrythm, which should be used
+             * to calculate the positions of the nodes.
+             *
+             *
+             * @sample highcharts/demo/treegraph-chart
+             *         Treegraph chart
+             *
+             * @type {'Walker'}
+             * @since next
+             * @default 'Walker'
+             * @product highcharts
+             */
+            layout: 'Walker',
+            /**
+             * Whether the first node should be placed on the opposite side of
+             * the plotArea. In default, the oldest child is positioned on the
+             * bottom (left in the inverted chart).
+             *
+             * @type {boolean}
+             * @since next
+             * @default false
+             * @product highcharts
+             *
+             * @sample highcharts/series-treegraph/reversed-nodes
+             *         Treegraph series with reversed nodes.
+             */
+            reversed: false,
+            /**
+             * @extends   plotOptions.series.marker
+             * @excluding enabled, enabledThreshold
+             */
+            marker: {
+                style: {
+                    cursor: 'pointer'
```

Comment:
And what if the node is not collapsible? Do we want to have `cursor: 'pointer'` for such nodes as well?